src/arviz_plots/plots/tracedensplot.py

```diff
@@ -0,0 +1,241 @@
+        plot_kwargs = {}
+    if pc_kwargs is None:
+        pc_kwargs = {}
+    else:
+        pc_kwargs = pc_kwargs.copy()
+
+    var_names = _var_names(var_names, dt.posterior.ds, filter_vars)
+
+    if var_names is None:
+        posterior = dt.posterior.ds
+    else:
+        posterior = dt.posterior.ds[var_names]
+
+    pc_kwargs.setdefault("aes", {"color": ["chain"]})
+
+    aux_dim_list = [dim for dim in posterior.dims if dim not in {"chain", "draw"}]
+
+    figsize, textsize, linewidth = scale_fig_size(
+        pc_kwargs.get("plot_grid_kws", {}).get("figsize", None),
+        rows=get_size_of_var(posterior, compact=compact),
+        cols=2,
+    )
+
+    plot_kwargs.setdefault("dens", {}).setdefault("lw", linewidth)
+    plot_kwargs.setdefault("trace", {}).setdefault("lw", linewidth)
+
+    if plot_collection is None:
+        if backend is None:
+            backend = rcParams["plot.backend"]
+        pc_kwargs.setdefault("cols", ["__column__"])
+        pc_kwargs.setdefault("plot_grid_kws", {"figsize": figsize})
+
+        if compact:
+            pc_kwargs.setdefault("rows", ["__variable__"])
+        else:
+            pc_kwargs.setdefault("rows", ["__variable__"] + aux_dim_list)
+
+        plot_collection = PlotCollection.grid(
+            posterior.expand_dims(__column__=2),
+            backend=backend,
+            **pc_kwargs,
+        )
+
+    if aes_map is None:
+        aes_map = {"trace": plot_collection.aes_set}
+
+    if labeller is None:
+        labeller = BaseLabeller()
+
+    density_dims, _, density_ignore = filter_aes(plot_collection, aes_map, "trace", sample_dims)
+
+    if compact:
+        density_dims += aux_dim_list
+
+    # dens
+    if kind == "kde":
+        density = posterior.azstats.kde(dims=density_dims, **dens_kwargs.get("density", {}))
+        plot_collection.map(
+            line_xy,
+            "dens",
+            data=density,
+            ignore_aes=density_ignore,
+            coords={"__column__": 0},
+            **plot_kwargs.get("dens", {}),
+        )
+
+    elif kind == "ecdf":
+        density = posterior.azstats.ecdf(dims=density_dims, **dens_kwargs.get("density", {}))
+        plot_collection.map(
+            ecdf_line,
+            "dens",
+            data=density,
+            ignore_aes=density_ignore,
+            coords={"__column__": 0},
+            **plot_kwargs.get("dens", {}),
+        )
+
+    # trace
+    plot_collection.map(
+        line,
+        "trace",
+        data=posterior,
+        ignore_aes=density_ignore,
+        coords={"__column__": 1},
+        **plot_kwargs.get("trace", {}),
+    )
+
+    ## aesthetics
+    # Remove yticks, only for KDEs
+    if kind == "kde":
+        _, _, yticks_dens_ignore = filter_aes(plot_collection, aes_map, "yticks_dens", sample_dims)
+
+        plot_collection.map(
+            remove_ticks,
+            "yticks_dens",
+            ignore_aes=yticks_dens_ignore,
+            coords={"__column__": 0},
+        )
```

Comment:
This doesn't return anything, so it should not be stored:

```suggestion
        plot_collection.map(
            remove_ticks,
            "yticks_dens",
            ignore_aes=yticks_dens_ignore,
            coords={"__column__": 0},
            store_artist=False,
            axis="y", # maybe also be explicit here?
        )
```

Side note, currently, using `None` as the 2nd argument (`fun_label`) uses `fun.__name__` as default label for the stored artist. Do you think using `None` should instead mean don't store it? Then we could remove the `store_artist` argument from .map